pyop2/openmp.py

```diff
@@ -206,96 +241,67 @@ def generate_code(self):
 
 class ParLoop(device.ParLoop, host.ParLoop):
 
+    def prepare_arglist(self, iterset, *args):
+        arglist = []
+        offset_args = []
+
+        if isinstance(iterset, Subset):
+            arglist.append(iterset._indices.ctypes.data)
+        for arg in self.args:
+            if arg._is_mat:
+                arglist.append(arg.data.handle.handle)
+            else:
+                for d in arg.data:
+                    arglist.append(d._data.ctypes.data)
+            if arg._is_indirect or arg._is_mat:
+                maps = as_tuple(arg.map, Map)
+                for map in maps:
+                    for m in map:
+                        arglist.append(m._values.ctypes.data)
+                        if m.iterset._extruded:
+                            offset_args.append(m.offset.ctypes.data)
+        for c in Const._definitions():
+            arglist.append(c._data.ctypes.data)
+
+        arglist.extend(offset_args)
+
+        if iterset._extruded:
+            region = self.iteration_region
+            if region is ON_BOTTOM:
+                arglist.append(0)
+                arglist.append(1)
+            elif region is ON_TOP:
+                arglist.append(iterset.layers - 2)
+                arglist.append(iterset.layers - 1)
+            elif region is ON_INTERIOR_FACETS:
+                arglist.append(0)
+                arglist.append(iterset.layers - 2)
+            else:
+                arglist.append(0)
+                arglist.append(iterset.layers - 1)
```

Comment:
But there's no explicit indication that these are the layer bounds!  Had the code been
```
arglist.append(("lower_bound", 0))
arglist.append(("upper_bound", 1))
```
then it would have been clear.  But it looks like the bounds are arbitrarily inserted into the middle of a list, and I guess this list is unpicked later on?